src/test/java/org/kohsuke/github/AppTest.java

```diff
@@ -291,7 +291,7 @@ public void testShouldFetchTeam() throws Exception {
         GHOrganization organization = gitHub.getOrganization(GITHUB_API_TEST_ORG);
         GHTeam teamByName = organization.getTeams().get("Core Developers");
 
-        GHTeam teamById = gitHub.getTeam(teamByName.getId());
+        GHTeam teamById = gitHub.getTeam((int) teamByName.getId());
```

Comment:
Right... so this means there will be build breaks for clients and we can't have that.

Reply:
That would mean we would have not to extend `GHObject`..
And thats actually really good idea..

Im not sure how bridge methods should work (but in java you cannot have 2 methods with same signature just different return type), but probably it just work at runtime..
At compile time it for sure gives error..

Reply:
If you want me to remove extension and implement node id in GHTeam without issues..

Reply:
@bitwiseman From code perspective I see that such "breaking" changes was already being mage few times (that changes are binary compatible because of that bridge methods)

See here
https://github.com/github-api/github-api/commit/b6e48cc4f9c7b13895673a7b4aa6399b7e236b1f
https://github.com/github-api/github-api/commit/edd9a2d5b6057a00afbe1d6c7cf329262713afc3

And GHObject was also moved from int to long not far away...
https://github.com/github-api/github-api/pull/388

So I think its safe enought to do it..